coordinator/statement_executor.go

```diff
@@ -1039,10 +1039,25 @@ func (e *StatementExecutor) executeShowTagValues(ctx *query.ExecutionContext, q
 		return err
 	}
 
-	// Get all shards for all retention policies.
+	// If measurements include retention policies
+	// only look at those policies
+	rps := make([]string, 0, len(di.RetentionPolicies))
+	// Collect retention policies if specified
+	for _, m := range q.Sources.Measurements() {
+		if len(m.RetentionPolicy) > 0 {
+			rps = append(rps, m.RetentionPolicy)
```

Comment:
What happens for multiple measurements with different specified retention policies? We take the union?

Reply:
Yes, that is correct.

Reply:
Prohibited multiple retention policies in SHOW TAG VALUES. Just no easy way to plumb the {shard, measurement} pair through the code everywhere.

Reply:
Would be good to have a test for the case we are prohibiting

Reply:
Test added.